BookingsApi/BookingsApi.Infrastructure.Services/AsynchronousProcesses/ParticipantUpdateToHearingAsynchronousProcess.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Linq;
+using BookingsApi.Common.Services;
+using BookingsApi.Domain;
+using BookingsApi.Infrastructure.Services.Publishers;
+using System.Threading.Tasks;
+using BookingsApi.Domain.Enumerations;
+using BookingsApi.Infrastructure.Services.IntegrationEvents.Events;
+
+namespace BookingsApi.Infrastructure.Services.AsynchronousProcesses
+{
+    public interface IParticipantUpdateToHearingAsynchronousProcess
+    {
+        Task Start(VideoHearing videoHearing);
+    }
+
+    public class ParticipantUpdateToHearingAsynchronousProcess : IParticipantUpdateToHearingAsynchronousProcess
+    {
+        private readonly IEventPublisherFactory _publisherFactory;
+
+        public ParticipantUpdateToHearingAsynchronousProcess(IEventPublisherFactory publisherFactory)
+        {
+            _publisherFactory = publisherFactory;
+        }
+
+        public async Task Start(VideoHearing videoHearing)
+        {
+            var participants = videoHearing.GetJudiciaryParticipants().ToList();
+        
+
+            switch (videoHearing.Status)
+            {
+                case BookingStatus.Created or BookingStatus.ConfirmedWithoutJudge:
+                    await _publisherFactory.Get(EventType.ParticipantAddedEvent).PublishAsync(videoHearing);
+                    await _publisherFactory.Get(EventType.HearingNotificationForJudiciaryParticipantEvent).PublishAsync(videoHearing);
+                    break;
+                case BookingStatus.Booked or BookingStatus.BookedWithoutJudge when participants.Exists(p => p.HearingRoleCode == JudiciaryParticipantHearingRoleCode.Judge):
+                    await _publisherFactory.Get(EventType.CreateConferenceEvent).PublishAsync(videoHearing);
+                    break;
+            }
```

Comment:
I think there's too much in this class, we only needed to send the messages from here, not have any business logic.

Reply:
`await _publisherFactory.Get(EventType.ParticipantAddedEvent).PublishAsync(videoHearing);
  await _publisherFactory.Get(EventType.HearingNotificationForJudiciaryParticipantEvent).PublishAsync(videoHearing);`
  Just these two lines called from this method PublishEventForNewJudiciaryParticipantsAsync

Reply:
sorry but I am not sure what you mean here. here there is only one switch case (not too much logic) as there is in the other processes with if then blocks.

Reply:
This is a cut and paste of the content of the PublishEventForNewJudiciaryParticipantsAsync, all that does is call this method now, so arguably that doesnt need to exist now.

This just needed to be two lines

await _publisherFactory.Get(EventType.ParticipantAddedEvent).PublishAsync(videoHearing);
                    await _publisherFactory.Get(EventType.HearingNotificationForJudiciaryParticipantEvent).PublishAsync(videoHearing);
                    
Called from within the case

Reply:
it is not just a copy and paste but it is adding the notification for judiciary participant and refactoring in the new async process as we discussed with Murali. this is what I have understood from our conversation